tests/src/test/scala/com/nvidia/spark/rapids/AnsiCastOpSuite.scala

```diff
@@ -466,6 +466,11 @@ class AnsiCastOpSuite extends GpuExpressionTestSuite {
     }
   }
 
+  testSparkResultsAreEqual("ansi_cast decimals to long",
+    generateValidValuesDecimalDF(Short.MinValue, Short.MaxValue, 18, 3), sparkConf) {
+    frame => testCastTo(DataTypes.LongType)(frame)
+  }
+
```

Comment:
Is it possible to test that we actually throw the expected exception?

Reply:
This test is only for the valid values. May be I need to add a test that tests just that an exception is thrown?

Reply:
We won't need to have a new case as you are throwing the same exception because this will be covered in `testCastFailsForBadInputs("ansi_cast overflow decimals to longs",..)`.
If we change the behavior to match the CPU, then we would need a new test.

Reply:
OK, then that test is already testing the exception that is thrown.